### PR TITLE
[CALCITE-6034] Add isAutoIncrement and isGenerated arguments to MetaColumn constructor

### PR DESCRIPTION
**Pull request by tjbanghart**

### CODE REVIEW

**Comment by tanclary** [comment]
nit: update commit message and PR description to be imperative (adds -> add)

File: core/src/main/java/org/apache/calcite/avatica/MetaImpl.java [hunk 1]
```diff
@@ -352,10 +352,11 @@ public static class MetaColumn implements Named {
     public final String scopeTable = null;
     public final Short sourceDataType = null;
     @ColumnNoNulls
-    public final String isAutoincrement = "";
+    public final String isAutoincrement;
     @ColumnNoNulls
-    public final String isGeneratedcolumn = "";
+    public final String isGeneratedcolumn;
 
+    @Deprecated // to be removed before 2.0
```
Comment (tanclary) [inline core/src/main/java/org/apache/calcite/avatica/MetaImpl.java:359]: I assume this is the constructor that was failing when building Calcite?

File: core/src/main/java/org/apache/calcite/avatica/MetaImpl.java [hunk 2]
```diff
@@ -370,6 +371,27 @@ public MetaColumn(
...
+        int dataType,
+        String typeName,
+        Integer columnSize,
+        Integer decimalDigits,
+        Integer numPrecRadix,
+        int nullable,
+        Integer charOctetLength,
+        int ordinalPosition,
+        String isNullable,
+        String isAutoincrement,
```
Comment (tanclary) [inline core/src/main/java/org/apache/calcite/avatica/MetaImpl.java:393]: is auto-increment one word?  Do you camelCase hyphenated words? Interesting..

Comment (tanclary) [reply core/src/main/java/org/apache/calcite/avatica/MetaImpl.java:393]: https://stackoverflow.com/questions/44170873/camel-casing-hyphenated-english-words-e-g-re-render#:~:text=1%20Answer&text=In%20camel%2Dcase%2C%20individual%20trailing,a%20single%20word%20would%20normally. this SO user makes a compelling case, sounds good to me

File: core/src/main/java/org/apache/calcite/avatica/MetaImpl.java [hunk 3]
```diff
@@ -383,11 +405,29 @@ public MetaColumn(
...
+      this.isGeneratedcolumn = isGeneratedcolumn;
     }
 
     public String getName() {
       return columnName;
     }
+
+    /** Returns a copy of this MetaColumn, overriding the value of {@code isAutoincrement}. */
+    @SuppressWarnings("unused") // called from Calcite
+    public MetaColumn withIsAutoIncrement(String isAutoincrement) {
```
Comment (tanclary) [inline core/src/main/java/org/apache/calcite/avatica/MetaImpl.java:418]: I would make it consistent either do AutoIncremeent or Autoincrement everywhere

**Review by tanclary — CHANGES_REQUESTED** [review]
Left a question and comment

File: core/src/main/java/org/apache/calcite/avatica/MetaImpl.java (hunk 3 above)
Comment (tjbanghart) [reply core/src/main/java/org/apache/calcite/avatica/MetaImpl.java:418]: lol, I thought about this for too long. I'll change it for consistency.

**Review by tanclary — APPROVED** [review]
LGTM, if you squash your commits I can merge : )

**Pull request merged**